app/models/workarea/flow_io/localized_price.rb

```diff
@@ -12,14 +12,24 @@ class LocalizedPrice
 
       validates_presence_of :regular
 
+      delegate :sku, :on_sale?, to: :local_item
+
+      def sell
+        if on_sale? && sale.present?
+          sale
+        else
+          regular
+        end
+      end
+
        # Creates a Pricing::Price from this local item
        # used in Pricing::Sku#find_price
        #
        # @return [Workarea::Pricing::Price]
        #
        def to_price
          Workarea::Pricing::Price.new(
-           sku: self.local_item.sku.clone, # clone the sku so this price isn't added to #prices on real record
+           sku: sku,
```

Comment:
performance seems to be OK on https://plugins-qa.demo.workarea.com with this added...I think it was the `#clone` causing all the trouble.

Reply:
Using sku causes this price to get persisted on the sku, mongoid sucks

Reply:
I don't see the return value of `#to_price` actually being saved anywhere, so I don't think this will have any harmful effects.

Reply:
According to the tests, replacing this with a proxy object that behaves like a `Pricing::Price` is also on the table, the only downside there is that any decorations made to `Pricing::Price` will also need to be made to this `FlowIo::PricingPriceProxy`. This might be a more performant way to go, but has a cognitive cost. I'm personally in favor of leaving it as a `Pricing::Price` since there's nothing that will ever save this object in the Flow code base, but if you think that could cause problems down the road, I'm all ears.

Reply:
Quick test:

```
irb(main):001:0> sku = Workarea::Pricing::Sku.sample
MONGODB | [13] localhost:27017 #2 | my_app_development.count | STARTED | {"count"=>"workarea_pricing_skus", "query"=>{}, "lsid"=>{"id"=><BSON::Binary:0x70359377325060 type=uuid data=0xef29dcb7317542fb...>}}
MONGODB | [13] localhost:27017 | my_app_development.count | SUCCEEDED | 0.004s
MONGODB | [14] localhost:27017 #2 | my_app_development.find | STARTED | {"find"=>"workarea_pricing_skus", "filter"=>{}, "skip"=>386, "sort"=>{"_id"=>1}, "limit"=>1, "singleBatch"=>true, "lsid"=>{"id"=><BSON::Binary:0x70359377325060 type=uuid data=0xef29dcb7317542fb...>}}
MONGODB | [14] localhost:27017 | my_app_development.find | SUCCEEDED | 0.009s
=> #<Workarea::Pricing::Sku _id: 992321835-X, created_at: 2020-03-03 16:55:39 UTC, updated_at: 2020-03-03 16:55:39 UTC, created_by_site_id: nil, active_segment_ids: {"en"=>[]}, active: {"en"=>true}, on_sale: false, discountable: true, tax_code: "001", msrp: {"cents"=>5826.0, "currency_iso"=>"USD"}>
irb(main):002:0> Workarea::Pricing::Price.new(sku: sku, min_quantity: 2, regular: 5.to_m)
=> #<Workarea::Pricing::Price _id: 5e68e3be021823835db80a41, created_at: nil, updated_at: nil, created_by_site_id: nil, active_segment_ids: {"en"=>[]}, active: {"en"=>true}, min_quantity: 2, regular: {:cents=>500.0, :currency_iso=>"USD"}, sale: nil, on_sale: false>
irb(main):003:0> Workarea::Pricing::Price.new(sku: sku, min_quantity: 2, regular: 5.to_m).persisted?
=> false
```

Another way to prevent the persisting would be to call `freeze` on the instance.